File: compiled_lang/java/e50.branch.java
import java.io.IOException;

class E50Branch {
    public static void main(String[] args) throws IOException {
        System.out.print("Input a character: ");
        char keypress = (char) System.in.read();

        // switch requires a single comparable value, not a pattern, so
        //  classify first and switch on the result
        int kind;
        if (Character.isUpperCase(keypress)) kind = 0;
        else if (Character.isLowerCase(keypress)) kind = 1;
        else if (Character.isDigit(keypress)) kind = 2;
        else kind = 3;

        switch (kind) {
            case 0: System.out.println("Uppercase letter"); break;
            case 1: System.out.println("Lowercase letter"); break;
            case 2: System.out.println("Digit"); break;
            default: System.out.println("Punctuation, whitespace, or other");
        }
    }
}

File: compiled_lang/go/src/e50.branch.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"unicode"
)

func main() {
	fmt.Print("Input a character: ")
	reader := bufio.NewReader(os.Stdin)
	b, _ := reader.ReadByte()
	keypress := rune(b)

	// a tagless switch - each case is its own boolean expression, Go's
	//  idiomatic stand-in for a pattern-matching multiway branch
	switch {
	case unicode.IsUpper(keypress):
		fmt.Println("Uppercase letter")
	case unicode.IsLower(keypress):
		fmt.Println("Lowercase letter")
	case unicode.IsDigit(keypress):
		fmt.Println("Digit")
	default:
		fmt.Println("Punctuation, whitespace, or other")
	}
}
